modules/azure/automationaccount.go

```diff
@@ -0,0 +1,342 @@
+	exists, err := AutomationAccountRunAsConnectionValidatesE(t, automationAccountrunAsAccountName, runAsConnectionType, runAsCertificateThumbprint, automationAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return exists
+}
+
+// AutomationAccountRunAsConnectionValidatesE indicates whether the specified Azure Automation Account RunAs Account exists.
+func AutomationAccountRunAsConnectionValidatesE(t testing.TestingT, automationAccountrunAsAccountName string, runAsConnectionType string, runAsCertificateThumbprint string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	runAsAccountConnection, err := GetAutomationAccountRunAsConnectionE(t, automationAccountrunAsAccountName, automationAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return *runAsAccountConnection.Name == automationAccountrunAsAccountName &&
+		*runAsAccountConnection.ConnectionProperties.ConnectionType.Name == runAsConnectionType &&
+		*runAsAccountConnection.ConnectionProperties.FieldDefinitionValues["CertificateThumbprint"] == runAsCertificateThumbprint, nil
+}
+
+// AutomationAccountDscAppliedSuccessfullyToVM indicates whether the specified Azure Automation Account compiled DSC has successfully been applied to the target VM.
+// This function would fail the test if there is an error.
+func AutomationAccountDscAppliedSuccessfullyToVM(t testing.TestingT, dscConfiguraitonName string, vmName string, automationAccountName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := AutomationAccountDscAppliedSuccessfullyToVME(t, dscConfiguraitonName, vmName, automationAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return exists
+}
+
+// AutomationAccountDscAppliedSuccessfullyToVME indicates whether the specified Azure Automation Account compiled DSC has successfully been applied to the target VM.
+func AutomationAccountDscAppliedSuccessfullyToVME(t testing.TestingT, dscConfiguraitonName string, vmName string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	dscNodeConfig, err := GetAutomationAccountDscNodeConfigurationE(t, dscConfiguraitonName, vmName, automationAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return *dscNodeConfig.Status == "Compliant", nil
+}
+
+/////////
+// Helper Methods for above checks
+/////////
+
+// GetAutomationAccountE returns the Azure Automation Account by name if it exists in the subscription
+func GetAutomationAccountE(t testing.TestingT, automationAccountName string, resourceGroupName string, subscriptionID string) (*automation.Account, error) {
```

Comment:
I would add a nonE version to those methods to open it up for general use

Reply:
Added nonE versions of the methods for general use.